### PR TITLE
add doc string for squad metric

### PR DESCRIPTION
**Pull request by smallv0221**
### PR types

Others 
### PR changes

Docs 
### Description

add doc string for squad metric

### CODE REVIEW

**Review by ZeyuChen — COMMENTED** [review]
LGTM

**Review by ZeyuChen — DISMISSED** [review]
LGTM

File: paddlenlp/metrics/squad.py [hunk 1]
```diff
@@ -23,34 +23,41 @@
...
-            Only useful when :obj:`version_2_with_negative` is :obj:`True`.
+        examples (list): List of raw squad-style data (see `run_squad.py 
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
+        features (list): List of processed squad-style features (see 
+            `run_squad.py <https://github.com/PaddlePaddle/PaddleNLP/blob/
+            develop/examples/machine_reading_comprehension/SQuAD/run_squad.py>`__ 
+            for more information).
+        predictions (tuple): The predictions of the model. should be a tuple
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: should -> Should

File: paddlenlp/metrics/squad.py [hunk 2]
```diff
@@ -23,34 +23,41 @@
...
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
+        features (list): List of processed squad-style features (see 
+            `run_squad.py <https://github.com/PaddlePaddle/PaddleNLP/blob/
+            develop/examples/machine_reading_comprehension/SQuAD/run_squad.py>`__ 
+            for more information).
+        predictions (tuple): The predictions of the model. should be a tuple 
+            of two list containing the start logits and the end logits. 
+        version_2_with_negative (bool, optional): If the dataset contains
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: If -> Whether

File: paddlenlp/metrics/squad.py [hunk 3]
```diff
@@ -357,6 +363,26 @@ def squad_evaluate(examples,
...
+        preds (dict): Dictionary of final predictions. Usually generated by 
+            `compute_prediction`.
+        na_probs (dict, optional): Dictionary of score_diffs of each example.
+            Used to decide if answer exits and compute best score_diff 
+            threshold of null. Default to None.
+        na_prob_thresh (float, optional): The threshold used to select the 
+            null answer. Default to 1.0.
+        is_whitespace_splited (bool, optional): If the predictions and references
+            can be tokenized by whitespace. Usually set True for English and 
+            False for Chinese. Default to True.
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: Default -> Defaults

File: paddlenlp/metrics/squad.py [hunk 4]
```diff
@@ -357,6 +363,26 @@ def squad_evaluate(examples,
...
+    Args:
+        examples (list): List of raw squad-style data (see `run_squad.py 
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
+        preds (dict): Dictionary of final predictions. Usually generated by 
+            `compute_prediction`.
+        na_probs (dict, optional): Dictionary of score_diffs of each example.
+            Used to decide if answer exits and compute best score_diff 
+            threshold of null. Default to None.
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: Default -> Defaults

File: paddlenlp/metrics/squad.py [hunk 5]
```diff
@@ -357,6 +363,26 @@ def squad_evaluate(examples,
...
+
+    Args:
+        examples (list): List of raw squad-style data (see `run_squad.py 
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
+        preds (dict): Dictionary of final predictions. Usually generated by 
+            `compute_prediction`.
+        na_probs (dict, optional): Dictionary of score_diffs of each example.
+            Used to decide if answer exits and compute best score_diff
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: 行尾不需要空格（换行已经代表空格了）

File: paddlenlp/metrics/squad.py [hunk 6]
```diff
@@ -23,34 +23,41 @@
...
-            The threshold used to select the null answer: if the best answer has a score that is less than the score of
-            the null answer minus this threshold, the null answer is selected for this example (note that the score of
-            the null answer for an example giving several features is the minimum of the scores for the null answer on
-            each feature: all features must be aligned on the fact they `want` to predict a null answer).
-
-            Only useful when :obj:`version_2_with_negative` is :obj:`True`.
+        examples (list): List of raw squad-style data (see `run_squad.py 
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py:40]: 感觉这里链接到run_squad.py代码对用户不太友好。

File: paddlenlp/metrics/squad.py [hunk 7]
```diff
@@ -23,34 +23,41 @@
...
+        features (list): List of processed squad-style features (see 
+            `run_squad.py <https://github.com/PaddlePaddle/PaddleNLP/blob/
+            develop/examples/machine_reading_comprehension/SQuAD/run_squad.py>`__ 
+            for more information).
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: 感觉这里链接到run_squad.py代码对用户不太友好，建议在文档中详细写一下。

File: paddlenlp/metrics/squad.py [hunk 8]
```diff
@@ -23,34 +23,41 @@
...
+        max_answer_length (int, optional): The maximum length of predicted answer. 
+            Defaults to 20.
+        null_score_diff_threshold (float, optional): The threshold used to select 
+            the null answer. Only useful when `version_2_with_negative` is True.
+            Defaults to 0.0.
+    
+    Returns:
+        A tuple of three dictionaries containing final selected answer, all n_best 
+        answers along with their probability and scores, and the score_diff of each 
+        example.
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py:60]: 感觉这里three dictionaries的具体结构也需要介绍一些

File: paddlenlp/metrics/squad.py [hunk 9]
```diff
@@ -357,6 +363,26 @@ def squad_evaluate(examples,
...
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
+        preds (dict): Dictionary of final predictions. Usually generated by 
+            `compute_prediction`.
+        na_probs (dict, optional): Dictionary of score_diffs of each example.
+            Used to decide if answer exits and compute best score_diff 
+            threshold of null. Default to None.
+        na_prob_thresh (float, optional): The threshold used to select the 
+            null answer. Default to 1.0.
+        is_whitespace_splited (bool, optional): If the predictions and references
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: If -> Whether

File: paddlenlp/metrics/squad.py [hunk 10]
```diff
@@ -357,6 +363,26 @@ def squad_evaluate(examples,
...
                    na_prob_thresh=1.0,
                    is_whitespace_splited=True):
+    '''
+    Computes and prints the f1 score and em score of input prediction.
+
+    Args:
+        examples (list): List of raw squad-style data (see `run_squad.py 
+            <https://github.com/PaddlePaddle/PaddleNLP/blob/develop/examples/
+            machine_reading_comprehension/SQuAD/run_squad.py>`__ for more 
+            information).
```
Comment (yingyibiao) [inline paddlenlp/metrics/squad.py]: 感觉这里链接到run_squad.py代码对用户不太友好。

File: paddlenlp/metrics/squad.py (hunk 7 above)
Comment (smallv0221) [reply paddlenlp/metrics/squad.py]: 这个metric的通用型其实不是很强，可以把它看做是example的一部分。依赖于咱们example的写法

File: paddlenlp/metrics/squad.py (hunk 8 above)
Comment (smallv0221) [reply paddlenlp/metrics/squad.py:60]: 这个结构不是在代码体现的么

Comment (smallv0221) [reply paddlenlp/metrics/squad.py:60]: 其实这三个也是squad中约定俗成的三个数据格式，这里应该假设用户对squad有基本了解

**Review by yingyibiao — APPROVED** [review]
LGTM
